File: 560.js

// Time Complexity : O(n^3)
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : Time Limit Exceeded

/**
Algorithm : Iterative/Brute force

Take two for loops and interate through array. Find all subarrays whose sum = k 
and increment count variable.
Return count.
**/

/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var subarraySum = function(nums, k) {
    var count = 0;
    
    for(var start=0; start < nums.length; start++){
        for(var end=start+1; end <= nums.length; end ++){
            var sum = 0;
            for(var i = start; i < end; i++){
                sum = sum + nums[i]
            }
            if(sum === k)
                count++;
            
        }
    }
  return count;  
};

File: 560.py
// Time Complexity : O(n ^ 2)
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : No //I tried the bruteforce approach and some testcases are failing
// Any problem you faced while coding this : Yes.


// Your code here along with comments explaining your approach


class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        count = 0
        temp_sum = 0
        for i in range(len(nums)):
            temp_sum = 0
            temp_sum += nums[i]
            if temp_sum == k:
                count += 1
            else:
                j = i + 1
                while (j < len(nums)):
                    if temp_sum + nums[j] > k :
                        break
                    elif temp_sum + nums[j] == k :
                        count += 1
                        break
                    else:
                        temp_sum += nums[j]
                        j += 1
        return count

                